Shared/Supporting Files/Views/SampleListView.swift

```diff
@@ -58,5 +70,34 @@ private extension SampleListView {
                 .animation(.easeOut(duration: 0.2), value: isShowingDescription)
             }
         }
+        
+        /// Bolds the first occurrence of substring within a given string using markdown.
+        /// - Parameters:
+        ///   - text: The `String` containing the substring.
+        ///   - substring: The substring to bold.
+        /// - Returns: The `String` with the bolded substring.
+        func boldSubstring(_ text: String, substring: String) -> String {
```

Comment:
I propose 2 options. Can you grab screenshots for both and add them to the screenshots in the description (that capture all name, description, and tags), so the next reviewers can vote on which path to go? 

Personally I see the bold text more than highlight, but I slightly lean towards highlight myself.

```swift
        // 1
        private func boldSubstring(_ text: String, substring: String) -> String {
            if let range = text.range(of: substring.trimmingCharacters(in: .whitespacesAndNewlines), options: .caseInsensitive) {
                var boldedText = text
                boldedText.replaceSubrange(range, with: "**" + text[range] + "**")
                return boldedText
            }
            return text
        }
        // 2
        private func attributedString(_ text: String, substring: String) -> AttributedString {
            var attributedString = AttributedString(text)
            if let range = attributedString.range(of: substring, options: .caseInsensitive) {
                attributedString[range].backgroundColor = .yellow.opacity(0.25)
            }
            return attributedString
        }
```

1. There is that issue with markdown syntax that contains leading/trailing whitespace. By trimming them off it works. Also the code can be much simplified.
2. We can also use the new [`AttributedString`](https://developer.apple.com/documentation/foundation/attributedstring) that Apple recommends to use starting from iOS 15. It works seamlessly with SwiftUI `Text` and is highly customizable.

- Mails uses yellow highlight; Maps, Contacts, System level search (Siri Suggestions) use bold text.
- I'll leave the method naming to Phil R.

Reply:
I am also in favor of using the `AttributedString` since Apple recommends using it and there are more of examples of it in their apps. It also shows up much better in the description than the bold text does. I will wait for others input, though.

Reply:
If you're going to stick with bolding (which may be fine, I don't know; I haven't looked at the rest of the PR), then unless I'm missing something, the implementation could be greatly simplified. Part of that simplification is moving the method to an extension of `String`:

```swift
private extension String {
    func boldingFirstOccurrence(of substring: String) -> String {
        guard let range = localizedStandardRange(of: substring) else {
            return self
        }
        return replacingCharacters(in: range, with: "**\(substring)**")
    }
}
```

Reply:
I agree, we can add the method as an extension to `String`.